### PR TITLE
Adding REGEX support for WARN and CRIT

### PR DESCRIPTION
**Pull request by kohenkatz**
I have an API that I check which returns multiple different status results that I want to consider WARN and CRIT. I figured that the easiest way to add this capability was to add Regular Expression support for `-W` and `-X`, since that allows for much more powerful usage.

Since the uppercase and lowercase letters were already used, I used the next letter in the alphabet for each argument. That can certainly be changed if needed.

### CODE REVIEW

**Comment by phrawzty** [comment]
Hello! Thanks for the PR - this is a good idea. 🎉 

Because this is a Nagios check, and Nagios servers tend to be old and cruddy, I test functionality against older versions of Ruby (currently 1.8.7, 1.9.3, and 2.4.0). If you can confirm that your new code works in those versions, I'd be happy to merge - it _looks_ fine, but it's good to test anyway. 😁

p.s. Thanks for also updating the help output and documentation - _super_ appreciated!

**Comment by phrawzty** [comment]
Ok, I went ahead and tested on the versions noted above, and it works (as expected, heh). Yay!

File: README.md [hunk 1]
```diff
@@ -25,6 +25,9 @@ Usage: ./check_http_json.rb -u <URI> -e <element> -w <warn> -c <crit>
     -W, --result_warn STRING         Warning if element is [string]. -C is required.
     -U, --result_unknown STRING      Unknown if element is [string]. -C is required.
     -C, --result_crit STRING         Critical if element is [string]. -W is required.
+    -X, --result_warn_regex REGEX    Warning if element matches REGEX. -C is required.
```
Comment (phrawzty) [inline README.md]: `--result_warn_regex` is part of the REGEX result group, so `--result_crit_regex` (aka `-D`) is required - not `-C`.

File: README.md [hunk 2]
```diff
@@ -25,6 +25,9 @@ Usage: ./check_http_json.rb -u <URI> -e <element> -w <warn> -c <crit>
...
+    -V, --result_unknown_regex REGEX Unknown if element matches REGEX. -C is required.
```
Comment (phrawzty) [inline README.md]: As above.

File: README.md [hunk 3]
```diff
@@ -25,6 +25,9 @@ Usage: ./check_http_json.rb -u <URI> -e <element> -w <warn> -c <crit>
...
+    -D, --result_crit_regex REGEX    Critical if element matches REGEX. -W is required.
```
Comment (phrawzty) [inline README.md]: As above.

**Comment by phrawzty** [comment]
> Since the uppercase and lowercase letters were already used, I used the next letter in the alphabet for each argument. That can certainly be changed if needed.

I totally understand the logic here, and normally I'd have no problem with this, except that `-d` already has a meaning and it has nothing to do with result levels. I'd rather not confuse the two. I see two options:
1. Use different upper-case single letters. This could cause collisions in the future.
1. Drop the single letter args entirely and just use the long args. This is the most future-proof, and to be honest, I wish I'd done it from the start with most of the options. 😕 

Unless you've got strong objections, let's go with № 2, ok?

**Comment by kohenkatz** [comment]
I was about to do that version testing, but since you got to it first, great.

I have no problem removing the short letter arguments.

**Comment by kohenkatz** [comment]
I will update the PR in a few hours.

**Comment by phrawzty** [comment]
Hi! This PR is now over a year old. I'm closing it - feel free to re-open if you want to move it forward. Thanks!

**Comment by kohenkatz** [comment]
Oops. Somehow this slipped completely off my radar. I will update the PR now.

**Comment by kohenkatz** [comment]
@phrawzty Are you able to reopen this? I am not able to do it, since you are the one who closed it.

**Comment by phrawzty** [comment]
Hi! There's a tiny merge conflict in README.md (literally one line). Can you please either fix it in your branch? You can just copy [from this branch](https://github.com/phrawzty/check_http_json/blob/kohenkatz-patch-1/README.md) if you want. 😄

**Comment by kohenkatz** [comment]
Fixed

**Comment by phrawzty** [comment]
It took as a year to get here, but we did it, haha.

**Pull request merged**